dawn/src/dawn/Optimizer/PassComputeStageExtents.cpp

```diff
@@ -27,71 +27,7 @@ PassComputeStageExtents::PassComputeStageExtents(OptimizerContext& context)
-    for(int i = numStages - 1; i >= 0; --i) {
-      iir::Stage& fromStage = *(stencil.getStage(i));
-      // If the stage has a global iterationspace set, we should never extend it since it is user
-      // defined where this computation should happen
-      if(std::any_of(fromStage.getIterationSpace().cbegin(), fromStage.getIterationSpace().cend(),
-                     [](const auto& p) { return p.has_value(); })) {
-        fromStage.setExtents(iir::Extents());
-        continue;
-      }
-
-      iir::Extents const& stageExtent = fromStage.getExtents();
-
-      // loop over all the input fields read in fromStage
-      for(const auto& fromFieldPair : fromStage.getFields()) {
-
-        const iir::Field& fromField = fromFieldPair.second;
-        auto&& fromFieldExtents = fromField.getExtents();
-
-        // notice that IO (if read happens before write) would also be a valid pattern
-        // to trigger the propagation of the stage extents, however this is not a legal
-        // pattern within a stage
-        // ===-----------------------------------------------------------------------------------===
-        //      Point one [ExtentComputationTODO]
-        // ===-----------------------------------------------------------------------------------===
-
-        iir::Extents fieldExtent = fromFieldExtents + stageExtent;
-
-        // check which (previous) stage computes the field (read in fromStage)
-        for(int j = i - 1; j >= 0; --j) {
-          iir::Stage& toStage = *(stencil.getStage(j));
-          // ===---------------------------------------------------------------------------------===
-          //      Point two [ExtentComputationTODO]
-          // ===---------------------------------------------------------------------------------===
-          auto fields = toStage.getFields();
-          auto it = std::find_if(fields.begin(), fields.end(),
-                                 [&](std::pair<int, iir::Field> const& pair) {
-                                   const auto& f = pair.second;
-                                   return (f.getIntend() != iir::Field::IntendKind::Input) &&
-                                          (f.getAccessID() == fromField.getAccessID());
-                                 });
-          if(it == fields.end())
-            continue;
-
-          // if found, add the (read) extent of the field as an extent of the stage
-          iir::Extents ext = toStage.getExtents();
-          ext.merge(fieldExtent);
-          // this pass is computing the redundant computation in the horizontal, therefore we
-          // nullify the vertical component of the stage
-          ext.resetVerticalExtent();
-          toStage.setExtents(ext);
-        }
-      }
-    }
-  }
-
-  for(const auto& MS : iterateIIROver<iir::MultiStage>(*(stencilInstantiation->getIIR()))) {
-    MS->update(iir::NodeUpdateType::levelAndTreeAbove);
-  }
-
+  stencilInstantiation->computeStageExtents();
```

Comment:
I think this is a dangerous thing to do. 

I assume the thought process is to not change any of the dependency set up, right?

But now we have a pass that can be called outside the pass framework which I don't really like.

I would prefer if the `Serializer` has a dependency on the Optimizer or if the IIR has an option `recomputeAllDerivedInfo` that you call when it is needed and remove the Pass. Option 2 could work with the StencilInstantiaion having the `computeStageExtents` call (or you could even push it down to the stage level, so every stage computes for itself).  Which is then initially called where we have the pass now and than in the recompute phase

Reply:
Went with Option 2:
- Renamed `computeStageExtents` to `computeDerivedInfo`
- Removed `PassComputeStageExtents` and replaced all occurences